### DISCUSSION TITLE
How can I compile any package specifically Termux.

### DISCUSSION THREAD

**Discussion by anshiq** [termux/termux-packages#10443] [Q&A] [answered]
Please help me with how can I add missing packages myself pull request to termux. I there is any specific termux compilation to add packages please help me.

**Comment by Biswa96** [comment, marked as answer]
This may help https://github.com/termux/termux-packages/wiki/Creating-new-package

**Reply by anshiq** [reply]
Means I have to just give the source code link, version, dependencies and other details in pull-up request. Termux team will automatically compile and add that package to the repo.

**Reply by xtkoba** [reply]
Remember that we are not bots. Thanks.

**Reply by ghost** [reply]
and we have to go through reviews about your pull request to ensure integrity and no errors were found. and you must meet the requirements before you open a PR because we will reject it if the format doesn't met
https://github.com/termux/termux-packages/blob/master/CONTRIBUTING.md

If this is the first time you're doing this we strongly recommend reading the [Termux developers wiki](https://github.com/termux/termux-packages/wiki)

**Reply by anshiq** [reply]
> Remember that we are not bots. Thanks.
@xtkoba 
What do you mean bro that you are not bots? I know that you are not bots. I am just curious about how you make a package readable for termux from the source code. I have seen many many pull-up requests of new packages in which you are only giving a link to the source code, version and dependencies. But in repo, you upload compiled package. I just want to know that compile process.

**Reply by Grimler91** [reply]
> But in repo, you upload compiled package

No, we upload a build recipe, and github actions builds and uploads the package through our build scripts ([build-package.sh](https://github.com/termux/termux-packages/blob/master/build-package.sh) and [scripts/build](https://github.com/termux/termux-packages/tree/master/scripts/build)), see [wiki](https://github.com/termux/termux-packages/wiki/Building-packages).

**Reply by anshiq** [reply]
So, with the help of these scripts you convert the source code to a package which is understandable to termux exactly.
